### PR TITLE
build: add skip-fullfiles and skip-packs option

### PR DESCRIPTION
**Pull request by rchiossi**
This patch adds the --skip-fullfiles options for the 'build' subcommand
of mixer. This command will prevent fullfiles from being generated
during 'build update'. Likewise, the --skip-packs option prevents zero
packs form being generated.

This PR addresses issue #352

### CODE REVIEW

File: mixin/build.go [hunk 1]
```diff
@@ -161,7 +161,7 @@ func buildMix(prepNeeded bool) error {
 		_ = os.Remove(mixFlagFile)
 		return err
 	}
-	err = b.BuildUpdate("", 0, "", false, true, false)
+	err = b.BuildUpdate("", 0, "", false, true, false, false, false)
```
Comment (tmarcu) [inline mixin/build.go]: Oh man, this function signature is just ridiculous now ❌ ahah. Can we please figure out a better way to call this, or change every single parameter to a named one so it's not insane to call in a future spot? FTFFF means nothing to even me at this point 😅

Comment (matthewrsj) [reply mixin/build.go]: Maybe a struct containing the configuration for this specific command that contains all the bools necessary?

Comment (rchiossi) [reply mixin/build.go]: it is! I wasn't sure if I wanted to add to the mess or just clean it up at once. I took the easy path. ;)

Probably a better approach would be to set those properties in the builder object.

Comment (tmarcu) [reply mixin/build.go]: I don't blame you, but I like the rest of the code so if you can make it pretty in this commit that'd be awesomeeeeee!

Comment (rchiossi) [reply mixin/build.go]: I just noticed that two of those parameters (prefixflag and keepChroots)  are not even used anymore....

Comment (tmarcu) [reply mixin/build.go]: 😆 😂

Comment (icarus-sparry) [reply mixin/build.go]: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis and the associated video is worth reviewing if you want to do better than just adding a struct, although if I was doing it I would just add a struct.

File: mixin/build.go [hunk 2]
```diff
@@ -161,7 +161,15 @@ func buildMix(prepNeeded bool) error {
...
-	err = b.BuildUpdate("", 0, "", false, true, false)
+	params := builder.UpdateParameters{
+		MinVersion:    0,
+		Format:        "",
+		Publish:       true,
+		SkipSigning:   false,
+		SkipFullfiles: false,
+		SkipPacks:     false,
+	}
+	err = b.BuildUpdate(params)
```
Comment (icarus-sparry) [inline mixin/build.go]: maybe
 ``` err := b.BuildUpdate(&builder.UpdateParameters{Publish: true})```
to use a literal struct, as there is just one field that is taking the non-default value in this case.

File: builder/builder.go [hunk 3]
```diff
@@ -84,6 +84,16 @@ type Builder struct {
 	UpstreamVerUint32 uint32
 }
 
+// UpdateParameters contains the configuration parameters for building an update
+type UpdateParameters struct {
```
Comment (icarus-sparry) [inline builder/builder.go]: Whilst the names are very good, it wouldn't hurt to add a comment for each field.

File: mixin/build.go (hunk 2 above)
Comment (rchiossi) [reply mixin/build.go]: ah nice! I didn't know that language feature.

File: builder/builder.go (hunk 3 above)
Comment (rchiossi) [reply builder/builder.go]: Sounds good. I'll update

**Comment by rchiossi** [comment]
Updated based on @icarus-sparry feedback.

File: builder/builder.go [hunk 4]
```diff
@@ -1538,15 +1543,25 @@ func (b *Builder) BuildUpdate(prefixflag string, minVersion int, format string,
 	return nil
 }
 
-func (b *Builder) buildUpdateContent(timer *stopWatch, mixVersion uint32, minVersion uint32, format uint32, skipSigning bool) error {
+func (b *Builder) buildUpdateContent(params UpdateParameters, timer *stopWatch) error {
 	var err error
 
+	// TODO: move this to parsing configuration / parameter time.
+	// TODO: should this be uint64?
```
Comment (tmarcu) [inline builder/builder.go]: yes.. but we need to update this all over the code to be s/uint32/uint64 or just uint.

Comment (rchiossi) [reply builder/builder.go]: this TODO comment just moved along with the code below, it is not my comment.

Comment (tmarcu) [reply builder/builder.go]: No prob :) Just reiterating so everyone knows I want to change this in the code at some point.

**Pull request merged**